src/Native/Unix/System.Native/pal_io.c

```diff
@@ -1315,7 +1353,8 @@ int32_t SystemNative_CopyFile(intptr_t sourceFd, const char* srcPath, const char
     if (!copied && CopyFile_ReadWrite(inFd, outFd) != 0)
     {
         tmpErrno = errno;
-        close(outFd);
+        while ((ret = flock(outFd, LOCK_UN)) < 0 && errno == EINTR);
+        while ((ret = close(outFd)) < 0 && errno == EINTR);
```

Comment:
Retrying `close(2)` in Linux is a race condition (please see the notes section in the man page for details).

Reply:
Thanks for pointing that out. In the last iteration it may not be necessary.